src/main/java/me/zeroeightsix/fiber/tree/ConfigNodeImpl.java

```diff
@@ -40,6 +44,23 @@ public String getComment() {
         return comment;
     }
 
+    @Nullable
+    @Override
+    public ConfigBranch getParent() {
+        return this.parent;
+    }
+
+    public void detach() {
+        this.parent = null;
+    }
```

Comment:
If people call `detach` without modifying the parent's child collection, `parent` will be 'out-of-sync'. Is there a way we can prevent this? Maybe do the removing from parent in this call (mind a concurrentmodificationexception)?